common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -285,12 +285,14 @@ def _setup_and_take_timed_exam(self, hide_after_due=False):
         self.assertTrue(self.courseware_page.is_timer_bar_present)
 
         self.courseware_page.stop_timed_exam()
+        self.courseware_page.wait_for_page()
         self.assertTrue(self.courseware_page.has_submitted_exam_message())
 
         LogoutPage(self.browser).visit()
 
-    @ddt.data(True, False)
-    def test_timed_exam_flow(self, hide_after_due):
+    #@ddt.data(True, False)
+    @flaky(max_runs=20, min_passes=20)
```

Comment:
FYI - I'm pretty sure you can use both decorators at once, as long as `ddt.data` is the one right above the method's signature.